test/blackbox-tests/test-cases/inline_tests/dune-file.t/run.t

```diff
@@ -30,3 +30,10 @@ package:
   Entering directory 'dune-file-user'
   inline_test_runner_foo_tests alias runtest
   414243
+
+We run the inline tests with a program that passes in specifc compile flags:
+
+  $ export OCAMLPATH=$PWD/_install/lib; dune runtest --root dune-file-compile-flags
```

Comment:
I'm not %100 sure what the best way to test this will be, but I think something like the following might work:

- Create a new test file under the `inline_tests` directory called something like `/compiler-flags.t/run.t`
- Set up a backend runner there that just works to detect whether certain conspicuous compiler flags have been set
- Set up one or two example targets that build the tests using different compiler flags

Do you have any thoughts for a more elegant test set up @jeremiedimino?

Reply:
That seems about right. You could also pass a flag that you are sure the compiler doesn't accept and observe the compilation failure.

BTW, the above test is using the `OCAMLPATH` trick to test that inline test backends work even when installed on the system. But since we already tested that, we don't need to use this method again and we can instead use a locally defined backend. So the shell command could just be: `dune runtest dune-file-compile-flags`.